### DISCUSSION TITLE
WIP - Comtrya 2022 roadmap

### DISCUSSION THREAD

**Discussion by icepuma** [comtrya/comtrya#209] [General]
# Roadmap

## OS specific support
* Add user management support (Groups / Deletion / Setup SSH keys / ...)
  * OpenBSD
  * NetBSD
  * DragonFly BSD
  * Linux
  * macOS
  * Windows
* Package providers
  * OpenBSD
  * FreeBSD (ports tree)
  * NetBSD (pkgsrc)
  * Ravenports (ravensw / ravenadm)
* Firewall rules

## Include variables
* Bitwarden

**Comment by martintc** [comment]
Off the top of my head, what I have in mind for the roadmap so far is as follow

- package provider for openbsd
- user add for openbsd
- user add for netbsd
- user add for dragonfly
- user add for linux
- user add for macos
- user add for windows
- package provider for FreeBSD ports tree
- package provider for NetBSD's pkgsrc
- package provider for Ravenports binary ravensw and ravenadm
- Figure out way to set up user credentials for ssh keys when using SSH 
- group management feature
- Probably should provide a user delete action
- Possibly a way to use comtrya to apply firewall rules

Those are at least the things off the top of my head that I would like to see implemented in Comtrya. They are not crazy advanced features, but features that would help form a solid base in configuration management.

**Reply by icepuma** [reply]
I moved all your input inside a more structured roadmap. Keep posting as I'll update the thingy.

**Reply by martintc** [reply]
Sounds good.

One thing I would like to figure it is how we can also use comtrya to set up the ssh keys for sshing into machines.

**Reply by icepuma** [reply]
Ah you mean: `comtrya` sets up the SSH key with the SSH server, so people can connect to the machine where `comtrya` is running?

**Reply by martintc** [reply]
Yea, it would be a pretty nice feature to have. How it should work, I am not sure. Maybe get the public keys from a private git repo?

**Reply by icepuma** [reply]
DNS TXT records :D
